webviz_subsurface/plugins/_parameter_analysis/models/parameters_model.py

```diff
@@ -206,39 +206,94 @@ def make_grouped_plot(
                 )
                 .update_xaxes(
                     matches=None,
                     fixedrange=True,
                     title=None,
-                    showticklabels=(len(parameters) < 20),
+                    showticklabels=len(parameters) <= 100,
+                    tickangle=0,
+                    tickfont_size=max((18 - (0.4 * len(parameters))), 8),
                 )
+                .update_yaxes(showticklabels=False)
                 .for_each_trace(
                     lambda t: t.update(
                         hoveron="violins",
-                        hoverinfo="name",
+                        hoverinfo="none",
                         meanline_visible=True,
                         orientation="h",
                         side="positive",
-                        width=3,
+                        width=2,
                         points=False,
                     )
                 )
                 .for_each_annotation(
                     lambda a: a.update(
-                        hovertext=a.text.split("=")[-1],
-                        text=(a.text.split("=")[-1]) if len(parameters) < 40 else "",
+                        text=(a.text.split("=")[-1]),
+                        visible=len(parameters) <= 42,
+                        font_size=max((18 - (0.4 * len(parameters))), 8),
                     )
                 )
             )
 
+        # Create invisible boxes used for hoverinfo on the violin plots
+        ensembles = df["ENSEMBLE"].unique()
+        hovertraces = []
+        for trace in fig["data"]:
+            parameter = trace["customdata"][0][0]
+            # check of parameter value to determine print formatter
+            value = abs(self.get_stat_value(parameter, ensembles[0], stat_column="Avg"))
+            form = ".1f" if value > 10 else ".2g"
+
+            hovertraces.append(
+                go.Scatter(
+                    x=[
+                        trace["x"].min(),
+                        trace["x"].min(),
+                        trace["x"].max(),
+                        trace["x"].max(),
+                    ],
+                    y=[" ", 1, 1, " "],
+                    xaxis=trace["xaxis"],
+                    yaxis=trace["yaxis"],
+                    mode="lines",
+                    fill="toself",
+                    marker_color="rgba(0, 0, 0, 0)",
+                    fillcolor="rgba(0, 0, 0, 0)",
+                    showlegend=False,
+                    text=(
+                        f"<b>{parameter}</b><br>"
+                        + "<br>".join(
+                            f"<b>{ens}:</b><br>"
+                            "Avg: "
+                            f"{self.get_stat_value(parameter, ens, stat_column='Avg'):{form}}<br>"
+                            "Std: "
+                            f"{self.get_stat_value(parameter, ens, stat_column='Stddev'):{form}}"
+                            for ens in ensembles
+                        )
```

Comment:
Feels unnatural to have one box with hoverinfo per ensemble, but still show data for both in each hoverinfo box (the boxes are equal.  Would rather view only the data for the trace (ensemble) you hover.

Reply:
Actually here I disagree 😉 The point of this parameter plotting is to be able to look at parameter updates from AHM. Having all the information in one hover box makes it easier to compare values from two iterations, instead of having to move back and forth between the traces.

In addition, when the distributions are similar, the boxes will overlap and it would be difficult to hover on the box in the back.

![image](https://user-images.githubusercontent.com/61694854/102650845-cf4a6f00-416b-11eb-88e6-17c17cbea2a5.png)

Reply:
I see your point, but is it then necessary with one box per trace, and not just one for the plot? 🤔 no problem to leave it as is for now